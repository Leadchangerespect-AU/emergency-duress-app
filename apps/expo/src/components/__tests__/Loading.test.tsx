import React from "react";
import { render, screen } from "@testing-library/react-native";

import LoadingScreen from "../Loading";

describe("LoadingScreen", () => {
  it("renders loading screen with all components", () => {
    render(<LoadingScreen />);

    // Check loading text
    expect(screen.getByText("Loading...")).toBeTruthy();

    // Check brand icon
    expect(screen.getByLabelText("brand icon")).toBeTruthy();

    // Check loading spinner
    expect(screen.getByLabelText("Loading spinner")).toBeTruthy();
  });

  it("configures brand icon correctly", () => {
    render(<LoadingScreen />);
    const brandIcon = screen.getByLabelText("brand icon");

    expect(brandIcon.props.source).toBeDefined();
    expect(brandIcon.props.contentFit).toBe("cover");
  });

  it("configures activity indicator correctly", () => {
    render(<LoadingScreen />);
    const spinner = screen.getByLabelText("Loading spinner");

    expect(spinner.props.size).toBe("large");
  });

  it("matches snapshot", () => {
    render(<LoadingScreen />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
