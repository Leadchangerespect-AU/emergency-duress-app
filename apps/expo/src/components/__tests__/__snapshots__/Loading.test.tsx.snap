// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingScreen matches snapshot 1`] = `
<View
  className="flex-1 items-center justify-center gap-4 bg-white"
>
  <ViewManagerAdapter_ExpoImage
    accessibilityLabel="brand icon"
    className="m-2 size-20"
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={
      [
        {
          "uri": "test-file-stub",
        },
      ]
    }
    style={{}}
    transition={null}
  />
  <Text
    accessibilityRole="header"
    className="animate-bounce text-primary"
    style={
      {
        "fontSize": 64,
        "fontWeight": "bold",
        "marginVertical": 21.44,
      }
    }
  >
    Loading...
  </Text>
  <ActivityIndicator
    aria-label="Loading spinner"
    className="text-primary"
    size="large"
  />
</View>
`;
